src/scripts/train_hallucination_detector.py

```diff
@@ -0,0 +1,186 @@
+"""Train hallucination detector.
+
+Usage:
+    uv run src/scripts/train_hallucination_detector.py <config_key>=<config_value> ...
+"""
+
+import logging
+import os
+
+import hydra
+import torch
+from datasets import Dataset, load_dataset
+from dotenv import load_dotenv
+from lettucedetect import HallucinationDataset, HallucinationSample
+from lettucedetect.models.evaluator import evaluate_model, print_metrics
+
+# from lettucedetect.models.evaluator import DataLoader
+from lettucedetect.models.trainer import Trainer
+from omegaconf import DictConfig
+from torch.utils.data import DataLoader
+from transformers import (
+    AutoModelForTokenClassification,
+    AutoTokenizer,
+    DataCollatorForTokenClassification,
+)
+
+from factuality_eval.train import format_dataset_to_ragtruth
+
+load_dotenv()
+
+
+def generate_lettucedetect_hallucination_samples(
+    dataset_split: Dataset,
+) -> list[HallucinationSample]:
+    """Generate hallucination samples for the LettuceDetect model.
+
+    Args:
+        dataset_split (Dataset): The dataset split to generate samples from.
```

Comment:
No need to include types in the docstring when we're using type hints:

```suggestion
        dataset_split: The dataset split to generate samples from.
```